addons/web/controllers/main.py

```diff
@@ -297,7 +297,16 @@ def set_cookie_and_redirect(redirect_url):
 
 def load_actions_from_ir_values(key, key2, models, meta):
     Values = request.session.model('ir.values')
-    actions = Values.get(key, key2, models, meta, request.context)
+    actions = []
+    for model in models:
+        res_id = False
+        if isinstance(model, (list, tuple)):
+            model, res_id = model
+        if key == 'default':
```

Comment:
I guess you kept the same signature for backwards-compatibility but I think we could use this opportunity to fix the ugly signature here too, given the fact that it's an internal helper method only called by the `/web/treeview/action` controller.

Wouldn't something like this be simpler and cleaner?

```python
def load_actions_from_ir_values(action_slot, model, res_id):
    actions = Values.get_actions(action_slot=action_slot, model=model,
        res_id=res_id, context=request.context)
    return [(id, name, clean_action(action)) for id, name, action in actions]
```